### DISCUSSION TITLE
How to use UPDATE FOR SELECT (postgres) using queryRaw or executeRaw?

### DISCUSSION THREAD

**Discussion by MiniSuperDev** [prisma/prisma#21335] [Q&A]
### Question

For example this query
```sql
BEGIN;
SELECT * FROM "counters" WHERE id = 1 FOR UPDATE;
UPDATE "counters" SET count = count + 1 WHERE id = 1;
COMMIT;
```

I tried
```ts
   prisma.$queryRaw`
BEGIN;
SELECT * FROM "counters" WHERE id = 1 FOR UPDATE;
UPDATE "counters" SET count = count + 1 WHERE id = 1;
COMMIT;
  `;
```

```sh
Invalid `prisma.$queryRaw()` invocation:

Raw query failed. Code: `42601`. Message: `db error: ERROR: cannot insert multiple commands into a prepared statement`
```

Thanks

___
More context:

I have a similar use case like this: https://on-systems.tech/blog/128-preventing-read-committed-sql-concurrency-errors/

Currently I use a mutex in my node main process, but I think is better to move this to the db.

If is possible can you show me a minimal example of how to implement the SELECT FOR UPDATE.

Thank you so much

**Comment by nurul3101** [comment]
Hello @MiniSuperDev 👋 

We have a related feature request here to natively support `SELECT FOR UPDATE`:

- https://github.com/prisma/prisma/issues/17136

The mentioned workaround suggests using `$queryRaw`, but it looks like Raw Query is not working for you.

Can you share your `counter` model? I can try to reproduce it on my end and investigate the issue.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
